Sources/Caching/KeyedDeferredValueStore.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  KeyedDeferredValueStore.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/12/25.
+
+import Foundation
+
+/// Some Task in which the value is Sendable and it can result in an error
+@_spi(Internal) public typealias AnyTask<T: Sendable> = Task<T, Error>
+
+/// Holds onto ``AnyTask`` objects by key, autoclearing on failure
+@_spi(Internal) public actor KeyedDeferredValueStore<H: Hashable, T: Sendable> {
```

Comment:
Another question here about if all of this needs to be `@_spi`-ed or if this can just be private for now. It seems like `RevenueCatUI` would end up using these indirectly through a `FileRepository`, correct? If that's the case. I would try to limit what is `@_spi`-ed 😊